src/main/java/org/jilt/internal/AbstractBuilderGenerator.java

```diff
@@ -333,13 +334,19 @@ protected final TypeVariableName mangleTypeParameter(TypeVariableName typeVariab
     private ParameterSpec setterParameterSpec(VariableElement attribute, TypeName parameterType) {
         ParameterSpec.Builder ret = ParameterSpec.builder(parameterType,
                 this.attributeSimpleName(attribute));
-        AnnotationMirror nullableAnnotation = this.firstAnnotationCalledNullable(attribute);
-        if (nullableAnnotation != null) {
-            ret.addAnnotation(AnnotationSpec.get(nullableAnnotation));
-        }
+        addAnnotations(ret, attribute.getAnnotationMirrors());
+        addAnnotations(ret, attribute.asType().getAnnotationMirrors());
         return ret.build();
     }
 
+    private void addAnnotations(final ParameterSpec.Builder ret, final List<? extends AnnotationMirror> annotationMirrors) {
+        for (final AnnotationMirror annotation : annotationMirrors) {
+            if (AnnotationUtils.isAnnotationAllowedInParameter(annotation)) {
+                ret.addAnnotation(AnnotationSpec.get(annotation));
+            }
+        }
+    }
+
```

Comment:
No `final` on local variables or method parameters:

```suggestion
    private void addAnnotations(ParameterSpec.Builder ret, List<? extends AnnotationMirror> annotationMirrors) {
        for (AnnotationMirror annotation : annotationMirrors) {
            if (AnnotationUtils.isAnnotationAllowedInParameter(annotation)) {
                ret.addAnnotation(AnnotationSpec.get(annotation));
            }
        }
    }
```